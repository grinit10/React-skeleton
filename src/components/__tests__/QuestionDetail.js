import { mapStateToProps, QuestionDetailDisplay } from '../QuestionDetail'
import renderer from 'react-test-renderer';
import React from 'react';

describe('The question details component testing', () => {

    describe(`The container element`, () => {
        describe(`mapStateToProps`, () => {
            it(`Should map the state to props correctly`, () => {
                const sampleQuestion = {
                    question_id: 42,
                    body: 'This is a question'
                };
                const appState = {
                    questions: [sampleQuestion]
                }
                const ownProps = {
                    question_id: 42
                }
                const containerState = mapStateToProps(appState, ownProps)
                expect(containerState).toEqual(sampleQuestion)
            })
        })
    })

    describe('The display element', () => {
        it(`Should not regress`, () => {
            const tree = renderer.create(
                <QuestionDetailDisplay title='Thanks'></QuestionDetailDisplay>
            )
            expect(tree.toJSON()).toMatchSnapshot();
        })
    })
})