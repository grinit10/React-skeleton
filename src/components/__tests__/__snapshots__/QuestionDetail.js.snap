// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The question details component testing The display element Should not regress 1`] = `
<div>
  <h3
    className="mb-2"
  >
    Thanks
  </h3>
  <div>
    <h4>
      Loading Question...
    </h4>
  </div>
</div>
`;
